### ISSUE TITLE
Add support for downloading publicly available pose datasets

### ISSUE THREAD

**Issue by niksirbi** [neuroinformatics-unit/movement#340]
This came up in discussions with @gchindemi.

There are several publicly available pose datasets - e.g. [CalMS21](https://paperswithcode.com/dataset/calms21) and [Rat7M](https://github.com/tqxli/rat7m). 

We could add some functionality for easily fetching these data in `movement`, by following a similar strategy to our [sample_data](https://movement.neuroinformatics.dev/getting_started/sample_data.html) module, i.e. download with `pooch`, cache locally, and (optionally) load into a [movement dataset](https://movement.neuroinformatics.dev/getting_started/movement_dataset.html).

Something like this would be useful for:
- us, because it would help us write tutorials/examples, and potentially to test and profile our code
- the community, especially developers of new methods, who could benefit from easy access to the various datasets and benchmarks during the prototyping phase

That said, we would have to think about somehow differentiating between publicly available datasets, and our own test data hosted on GIN (see also #237).

**Comment by niksirbi** [comment]
Clarifications:
- this would concern not only pose datasets, but also perhaps data with bounding boxes or segmentations masks (pending #301).
- we needn't do it all at once, we should start with one dataset, to figure out the interface, and then proceed from there.

**Comment by gchindemi** [comment]
I think this could be very useful. I implemented this functionality for the CalMS21 dataset in our social behavior analysis package [LISBET](https://github.com/BelloneLab/lisbet). I would be happy to contribute it to `movement`, after you decide how to expose public datasets.

**Comment by niksirbi** [comment]
Thanks @gchindemi, I'll keep you in the loop about this.

**Comment by SkepticRaven** [comment]
Keep me in the loop, too. The group I'm in has released a handful of big pose datasets including the [MABe22 mouse triplets](https://www.aicrowd.com/challenges/multi-agent-behavior-challenge-2022/problems/mabe-2022-mouse-triplets) and our [open field strain survey](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi%3A10.7910%2FDVN%2FSAPNJG), with more on the horizon.

**Comment by demoncoder-crypto** [comment]
I have given a draft solution for this

**Comment by lochhh** [comment]
Atm both [load_bboxes.from_file](https://movement.neuroinformatics.dev/api/movement.io.load_bboxes.from_file.html#movement.io.load_bboxes.from_file) and [load_poses.from_file](https://movement.neuroinformatics.dev/api/movement.io.load_poses.from_file.html#movement.io.load_poses.from_file) take `file_path` as input. As a first step to loading public datasets, we could extend this to support both local file paths and URLs. These URLs should be files from movement-supported software, so that we can continue using the software-specific loaders.

To quickly test, I tried to import [Lightning Pose data from the EKS repository](https://github.com/paninski-lab/eks/tree/main/data/mirror-mouse)
```python
import tempfile

import requests
from movement.io import load_poses

src = "https://raw.githubusercontent.com/paninski-lab/eks/refs/heads/main/data/mirror-mouse/180607_004.train_frames%3D75.rng%3D0.csv"
response = requests.get(src)
response.raise_for_status()
content = response.content
with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as tmp_file:
    tmp_file.write(content)
    tmp_file_path = tmp_file.name
load_poses.from_file(tmp_file_path, source_software="LightningPose")
```

**Comment by niksirbi** [comment]
Cool idea. Would the logic be as follows?

If path is supplied as URL, try to first download from that URL to a temp file, and if that succeeds call the `from_file` function again with that file path?

**Comment by lochhh** [comment]
Yes, I could see this bit being implemented as a condition in `from_file`, i.e. if file path is a URL, download the file. We should decide if we want to keep the file and download it to .movement. Otherwise, we need an additional step to `os.remove` or `Path.unlink()` the file.

**Comment by niksirbi** [comment]
In some cases it will be beneficial to keep the file in `~/.movement/data`, because that folder is cached in CI. This would come in handy in cases where we want to used the public data in examples, i.e. caching those files would speed up docs build in CI.  That said, if we do that we should put the files in a separate subfolder, i.e. don't mix it with data coming from our GIN.

Btw, I think we probably shouldn't bother adding this URL reading capacity to every `from_{software}_file` function, adding it just to the 2 general `from_file` functions would be easier.

**Comment by niksirbi** [comment]
I am now dreaming of the day when I will be able to just paste a URL in our `napari` loader widget, and the file will open in `napari` 😉

**Comment by lochhh** [comment]
> Btw, I think we probably shouldn't bother adding this URL reading capacity to every `from_{software}_file` function, adding it just to the 2 general `from_file` functions would be easier.

Agreed!

**Comment by Tushar7012** [comment]
Hi @niksirbi 

I've read through the entire discussion and would like to help.

I noticed PR #515 was linked to this issue but the status moved back to Triage in December. Is that PR still being worked on, or would it be helpful for someone to take over / start fresh?

**My understanding of the agreed approach:**
1. Add URL detection to the 2 general [from_file](cci:1://file:///d:/Agentic_AI/Gssoc_NLU/movement/movement/roi/base.py:592:4-641:13) functions
2. If URL provided → download to temp file → call [from_file](cci:1://file:///d:/Agentic_AI/Gssoc_NLU/movement/movement/roi/base.py:592:4-641:13) with local path
3. Cache in `~/.movement/data/public_datasets/` (separate from GIN data)

Happy to contribute if there's an opportunity!
